### PR TITLE
Feat: add iterable_merge function

### PR DESCRIPTION
**Pull request by jdecool**
Add `iterable_merge` function.

### CODE REVIEW

File: src/iterable-functions.php [hunk 1]
```diff
@@ -28,6 +28,21 @@ function iterable_map(iterable $iterable, callable $mapper): iterable
...
+ * Merge one or more iterables
+ *
+ * @param iterable<TKey, TValue> $iterable
+ * @param iterable<iterable<TKey, TValue>> ...$args
+ * @return iterable<TKey, TValue>
+ *
+ * @template TKey
+ * @template TValue
+ */
+function iterable_merge(iterable $iterable, iterable ...$args): iterable
```
Comment (bpolaszek) [inline src/iterable-functions.php]: As of PHP 7.4, `array_merge()` can be used without arguments (it basically returns an empty array). I'd suggest doing the same here.

**Review by bpolaszek — CHANGES_REQUESTED** [review]
Hi @jdecool,

Thanks for your contribution & sorry for the late reply!

The following case does not succeed:

```php
$array1 = ['bar', 'baz'];
$array2 = ['bat', 'baz'];
$generator1 = fn () => yield from $array1;
$generator2 = fn () => yield from $array2;

expect(iterable_merge($generator1(), $generator2())->asArray())->toEqual(array_merge($array1, $array2));
```

**Comment by jdecool** [comment]
No problem @bpolaszek 

I've just pushed some updates.

**Comment by bpolaszek** [comment]
Sounds good to me 🙂 

Can you please fix CI + advertise it in `README.md`?

Thanks! 🙏

**Comment by jdecool** [comment]
> Can you please fix CI + advertise it in README.md?

Sure, everything should be OK now.

**Comment by jdecool** [comment]
😮 it fails. 

Sorry i’ve missed one check

**Comment by jdecool** [comment]
Everything is fine now is the CI  😌

**Review by bpolaszek — APPROVED** [review]

**Comment by bpolaszek** [comment]
Thanks @jdecool, your contribution is very much appreciated! 👍

File: src/iterable-functions.php [hunk 2]
```diff
@@ -28,6 +29,21 @@ function iterable_map(iterable $iterable, callable $mapper): iterable
     return iterable($iterable)->map($mapper);
 }
 
+/**
+ * Merge iterables
+ *
+ * @param iterable<TKey, TValue> ...$args
```
Comment (simPod) [inline src/iterable-functions.php:35]: BTW I think this template will not work for merging iterables with different value types.

Also it seems, there's no support for typing it properly https://github.com/phpstan/phpstan/issues/3814#issuecomment-684779628

**Comment by JoshuaBehrens** [comment]
Nice feature :)

**Pull request merged**